src/poprox_recommender/embedders/article.py

```diff
@@ -34,6 +36,10 @@ def __init__(self, model: ArticleEmbeddingModel, tokenizer: PreTrainedTokenizer,
         self.embedding_cache = {}
 
     def __call__(self, article_set: ArticleSet) -> ArticleSet:
+        if not article_set.articles:
+            article_set.embeddings = th.zeros((0, self.model.embedding_size))  # type: ignore
+            return article_set
```

Comment:
While we don't call the profile with empty sets, this just makes this model a little more robust (although downstream components will break when this condition is triggered).